### PR TITLE
adds abstract Class Piece

### PR DESCRIPTION
**Pull request by Varunalingam**
adds package org.dwoc.khet.gameControl

adds abstract Class Piece in org.dwoc.khet.gameControl

### CODE REVIEW

File: app/src/main/java/org/dwoc/khet/gameControl/Piece.kt [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+ */
+
```
Comment (bestinbthomas) [inline app/src/main/java/org/dwoc/khet/gameControl/Piece.kt]: can you remove the Licence from here

**Review by bestinbthomas — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/dwoc/khet/gameControl/Piece.kt [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  {
+    this.PosX = PosX
+    this.PosY = PosY
+  }
+
+  //Function RotatePiece will Rotate the Piece's Orientation by the Said Amount
+  fun RotatePiece (amount : Int)
+  {
+     var value: Int = amount + Orientation.ordinal + 4
+    value %= 4
```
Comment (bestinbthomas) [inline app/src/main/java/org/dwoc/khet/gameControl/Piece.kt:30]: account for negative as well. if it becomes -1 set it to 3

File: app/src/main/java/org/dwoc/khet/gameControl/Piece.kt [hunk 3]
```diff
@@ -0,0 +1,31 @@
+package org.dwoc.khet.gameControl
+
+import org.dwoc.khet.models.enums.Direction
+import org.dwoc.khet.models.enums.Player
+
+// Each Types of Piece will Extend this Piece Class
+abstract class Piece(val PlayerId : Player, var PosX : Int, var PosY : Int, var Orientation : Direction)
+{
+
+  //Function MovePiece will set the PosX and PosY of the Piece
```
Comment (bestinbthomas) [inline app/src/main/java/org/dwoc/khet/gameControl/Piece.kt]:
```suggestion
  /**
  * MovePiece will set the PosX and PosY of the Piece
  * @param PosX
  * @param PosY
  */
```
can you change the comments to javadoc style coments

**Review by bestinbthomas — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/dwoc/khet/gameControl/Piece.kt (hunk 2 above)
Comment (Varunalingam) [reply app/src/main/java/org/dwoc/khet/gameControl/Piece.kt:21]: The + 4 in the end will make it positive then we will be able to use the %4 to get the remainder!

Comment (Varunalingam) [reply app/src/main/java/org/dwoc/khet/gameControl/Piece.kt:30]: The + 4 in line 20 after the Orientation.ordinal will account the negative 1 and makes it positive then the %4 will give values in range 0 to 3

**Review by bestinbthomas — APPROVED** [review]

**Pull request merged**